### PR TITLE
[RFC][Doc] add a page describe actor execution order.

### LINKED ISSUE #23342: Write documentation on task ordering for actors

### PR DESCRIPTION
**Pull request by scv119**
## Why are these changes needed?

The execution order is kinda of short. Should we merge this with fault-tolerance, or expand it a bit more with examples?
Also we don't guarantee orders if tasks are submitted by different workers. Not sure if we should put it here as well.
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/ray-core/actors/task-orders.rst [hunk 1]
```diff
@@ -0,0 +1,13 @@
+Actor Tasks Execution Orders
+============================
+
+Synchronous, single-threaded actors execute tasks according
+to their initial submission order. Any submitted task will not
```
Comment (ericl) [inline doc/source/ray-core/actors/task-orders.rst:5]: Add some code examples? I think it's also worth noting that tasks can get reordered between different submitters. For example, if they are waiting on object argument resolution.

File: doc/source/ray-core/actors/task-orders.rst [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+execute until previously submitted tasks have finished execution.
+Note that actor task execution could fail. Upon failure, the following
+tasks submitted to the same actor will not execute until the failed
+task is successfully retried or the retry is exhausted.
+
+:ref:`Async or threaded actors <async-actors>` does not guarantee the
+task execution order. For example, the system might execute a task
```
Comment (ericl) [inline doc/source/ray-core/actors/task-orders.rst:12]: Didn't you add ways to configure this? Or is it always unordered?

Comment (scv119) [reply doc/source/ray-core/actors/task-orders.rst:12]: https://sourcegraph.com/github.com/ray-project/ray/-/blob/python/ray/_raylet.pyx?L1607 today it's always unordered.

**Comment by scv119** [comment]
ready for another round of feedbacks!

File: doc/source/ray-core/actors/task-orders.rst [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+        # Output: 3. The first submitted task is executed later.
+        print(ray.get(value0))
+        # Output: 2. The later submitted task is executed first.
+        print(ray.get(value1))
+
+
+Asynchronous or Threaded Actor
+------------------------------
+:ref:`Asynchronous or threaded actors <async-actors>` does not guarantee the
```
Comment (ericl) [inline doc/source/ray-core/actors/task-orders.rst:87]:
```suggestion
:ref:`Asynchronous or threaded actors <async-actors>` do not guarantee the
```

File: doc/source/ray-core/actors/task-orders.rst [hunk 4]
```diff
@@ -0,0 +1,124 @@
+Actor Tasks Execution Orders
+============================
+
+Synchronous, Single-Threaded Actor
+----------------------------------
+In Ray, an actor receives tasks from multiple submitters (including driver and workers).
+For tasks received from the same submitter, synchronous, single-threaded actor executes
```
Comment (ericl) [inline doc/source/ray-core/actors/task-orders.rst:7]:
```suggestion
For tasks received from the same submitter, a synchronous, single-threaded actor executes
```

File: doc/source/ray-core/actors/task-orders.rst [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+them following the submission order.
+In other words, any task will not be executed until previously submitted tasks from
```
Comment (ericl) [inline doc/source/ray-core/actors/task-orders.rst:9]:
```suggestion
In other words, a given task will not be executed until previously submitted tasks from
```

**Review by ericl — APPROVED** [review]
Looks awesome. Just a couple bits.

**Pull request merged**